Error in REST API when fetching all discovered channels on win32 tester

I got the following error on the Windows 32-bit tester:

```
======================================================================
FAIL: Testing whether the API returns inserted channels when fetching discovered channels
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Test\test_as_server.py", line 81, in check
    result = fun(*argv, **kwargs)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Core\Utilities\twisted_thread.py", line 151, in errback
    failure.raiseException()
  File "C:\Python27\lib\site-packages\twisted\internet\defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win32\tribler\Tribler\Test\Core\Modules\RestApi\test_channels_endpoints.py", line 79, in verify_channels
    self.assertEqual(channels_json['channels'][i]['name'], 'Test channel %d' % i)
AssertionError: u'Test channel 2' != 'Test channel 0'
-------------------- >> begin captured logging << --------------------
```

Apparently, we cannot make any assumptions on the order of incoming data. We should remove this check.

--- a/Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py
+++ b/Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py
@@ -72,13 +72,6 @@ class TestChannelsEndpoint(AbstractTestChannelsEndpoint):
         expected_json = {u'channels': []}
         return self.do_request('channels/discovered', expected_code=200, expected_json=expected_json)
 
-    def verify_channels(self, channels):
-        channels_json = json.loads(channels)
-        self.assertEqual(len(channels_json['channels']), 10)
-        for i in range(len(channels_json['channels'])):
-            self.assertEqual(channels_json['channels'][i]['name'], 'Test channel %d' % i)
-            self.assertEqual(channels_json['channels'][i]['description'], 'Test description %d' % i)
-
     @deferred(timeout=10)
     def test_get_discovered_channels(self):
         """
@@ -88,7 +81,14 @@ class TestChannelsEndpoint(AbstractTestChannelsEndpoint):
         for i in range(0, 10):
             self.insert_channel_in_db('rand%d' % i, 42 + i, 'Test channel %d' % i, 'Test description %d' % i)
 
-        return self.do_request('channels/discovered', expected_code=200).addCallback(self.verify_channels)
+        def verify_channels(channels):
+            channels_json = json.loads(channels)['channels']
+            self.assertEqual(len(channels_json), 10)
+            channels_json = sorted(channels_json, key=lambda channel: channel['name'])
+            for ind in xrange(len(channels_json)):
+                self.assertEqual(channels_json[ind]['name'], 'Test channel %d' % ind)
+
+        return self.do_request('channels/discovered', expected_code=200).addCallback(verify_channels)
 
 
 class TestChannelTorrentsEndpoint(AbstractTestChannelsEndpoint):

